syntax = "proto3";

option objc_class_prefix = "RTG";

package social_media;

service UsersService {
  rpc GetUser(GetUserRequest) returns (GetUserResponse) {}
  rpc GetUserSummaries(GetUserSummariesRequest) returns (GetUserSummariesResponse) {}
}

message GetUserSummariesRequest {
}

message GetUserSummariesResponse {
  int32 count = 1;
  repeated GetUserSummaryResponse items = 2;

}

message GetUserSummaryResponse {
  User user = 1;
  UserFollowersResponse followers = 2;
}

message UserFollowersResponse {
  int32 count = 1;
  repeated UserFollowerResponse items = 2 ;
}

message UserFollowerResponse {
  int32 follower_id = 1;
  string username = 2;
  string email = 3;
}

message GetUserRequest {
  int32 user_id = 1;
}

message User {
  int32 user_id = 1;
  string username = 2;
  string email = 3;
  uint64 join_date_timestamp = 4;
  string profile_picture_url = 5;
  string bio = 6;
}


message GetUserResponse {
  bool found = 1;
  optional User user = 2;
}

